rstudio/Dockerfile

```diff
@@ -39,78 +39,124 @@
 
-# Install R
-# Since this Dockerfile uses the CML Python base image, it is necessary to install R. This will
-# not install the # cdsw package used for managing experiments and models with R sessions. Given
-# that models and # experiments don't use R-Studio as the editor, this is not a problem. Just make
-# sure to keep consistency in R versions. The current CML R Runtime uses R 4.0.4, so that is the
-# target version to install
-RUN apt-get install -y r-base-core=4.0.4-1.2004.0
-
 # Install R Studio
-RUN curl -O https://download2.rstudio.org/server/bionic/amd64/rstudio-server-2022.07.0-548-amd64.deb
-RUN gdebi -n rstudio-server-2022.07.0-548-amd64.deb
+RUN curl -O https://download2.rstudio.org/server/focal/amd64/rstudio-server-2023.09.1-494-amd64.deb
+RUN gdebi -n rstudio-server-2023.09.1-494-amd64.deb
+
 
 # Configure RStudio to work in CML
 # RStudio needs some changes made to the default configuration to work in CML, e.g.
 # port and host info. This is set in the two additonal files that are added to the image.
-
 RUN chown -R cdsw:cdsw /var/lib/rstudio-server && chmod -R 777 /var/lib/rstudio-server
 COPY rserver.conf /etc/rstudio/rserver.conf
 COPY rstudio-cml /usr/local/bin/rstudio-cml
 RUN chmod +x /usr/local/bin/rstudio-cml
-
 # Create a Symlink to the default editor launcher
 # This is the main requirement to get CML to launch a different editor. Create a symlink
 # from the editors launcher to /usr/local/bin/ml-runtime-editor
 RUN ln -s /usr/local/bin/rstudio-cml /usr/local/bin/ml-runtime-editor
+RUN export PATH=$PATH:/home/cdsw/R/x86_64-pc-linux-gnu-library/4.1
+USER root
+# MLX-50 Allow the engine image to be run as user cdsw upon startup.
+# To do this we must 'chmod 777' /etc related directories.
+RUN for i in /etc /var /etc/alternatives /var/lib/dpkg/ /home /home/dataiku /runtime-addons; do \
+ if [ -d ${i} ]; then chmod 777 ${i}; fi; \
+ done
+# DSE-11162 Base engine changes to support runtime addons.
+# To do this, we must 'chown cdsw' some directories.
+# Make root directory read/writable by cdsw user
+RUN chown cdsw /
+RUN chown cdsw /
+# Add permissions to cdsw user recursively to the common directories that gets touched as part
+# of the engine startup process.
+RUN for i in /bin /var /etc /opt /var/lib/dpkg/ /sbin /usr /home /home/dataiku /runtime-addons; do \
+ if [ -d ${i} ]; then \
+   chown cdsw ${i}; \
+   find ${i} -type d -exec chown cdsw {} +; \
+ fi; \
+ done
 
-# Install additional R dependencies
-COPY r-runtime-dependencies.txt /build/
-
-RUN \
-    apt-get update && \
-    cat /build/r-runtime-dependencies.txt | \
-      sed '/^$/d; /^#/d; s/#.*$//' | \
-      xargs apt-get install -y --no-install-recommends && \
-    rm -rf /var/lib/apt/lists/* && \
-    rm -rf /build
-
-# Change permissions to allow env variables to be pushed to Renviron.site file
-# when the rstudio-cml startup file is run
-RUN chmod -R 666 /etc/R/Renviron.site
```

Comment:
Why don't you need this anymore?

Reply:
This path doesn't exist in the new build, it seems.